chore(gatsby-cli): friendly docs for gatsby-cli new command (#17549)
gatsbyjs/gatsby: packages/gatsby-cli/README.md

### `new`

```bash
gatsby new [<site-name> [<starter-url>]]
```

#### Arguments

- **`site-name` :** Your Gatsby site name along with the path should be pass here`

- **`starter-url` :** Mention the gatsby starter pack url or local path here.
  See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more.
  `Default` starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default).

> **NOTE :** Dont pass `.` or `./` or `` _(space)_ in the `site-name`. It will throw error of specifying name and it will stop the operation

#### Examples

> - `gatsby new MY_AWESOME_GATSBY_SITE` : it will create a gatsby site with `MY_AWESOME_GATSBY_SITE` name in the current working directory using the default starter

Reviewer: please add one ` to code markup - the suggestion not display it right
```suggestion
- create a gatsby site with `my-awesome-site` name in the current working directory using the default starter:
``bash
gatsby new my-awesome-site
``
```

> - `gatsby new MY_AWESOME_GATSBY_BLOG_SITE https://github.com/gatsbyjs/gatsby-starter-blog` : It will create a gatbsy site with name `MY_AWESOME_GATSBY_BLOG_SITE` and using [gatsby-starter-blog](https://www.gatsbyjs.org/starters/gatsbyjs/gatsby-starter-blog/) starter pack

Reviewer: ```suggestion
- create a gatbsy site with name `my-awesome-blow-site` and using [gatsby-starter-blog](https://www.gatsbyjs.org/starters/gatsbyjs/gatsby-starter-blog/) starter pack:
``bash
gatsby new my-awesome-blow-site https://github.com/gatsbyjs/gatsby-starter-blog
``
```

> - `gatsby new https://github.com/gatsbyjs/gatsby-starter-blog` : It will result in error and will stop the command with a message : _It looks like you forgot to add a name for your new project. Try running instead "gatsby new new-gatsby-project https://github.com/gatsbyjs/gatsby-starter-blog"_

Reviewer: i would not suggest error messages as examples. insted do here the interactive

```suggestion
- If you leave all the both of the argument, then CLI will then run a interactive shell asking for these inputs:
``bash
gatsby new
? What is your project called? › my-gatsby-project  
? What starter would you like to use? › - Use arrow-keys. Return to submit. 
❯  gatsby-starter-default
   gatsby-starter-hello-world
   gatsby-starter-blog
   (Use a different starter)
``                                                                                                                
```

Author: What about the error message statement

Reviewer: examples are here to show "how to use it" and not to show "how you use it wrong". i would not put the error message. to use the cli you don't need to know when an error check is available.

Author: Yeah I removed them

Reviewer: 👍

Refer [starters-docs](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/starters.md) for more details

> #### Interactive prompting

Reviewer: ```suggestion
```

> If you leave all the both of the argument, then CLI will then run a interactive shell asking for these inputs.

Reviewer: ```suggestion
```

### `new`

```bash
gatsby new [<site-name> [<starter-url>]]
```

#### Arguments

| Argument    | Description                                                                                                                                                                                                                                         |
| ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| site-name   | Your Gatsby site name along with the path should be pass here                                                                                                                                                                                       |

> | starter-url | Mention the gatsby starter pack url or local path here. See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more. `Default` starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default). |

Reviewer: ```suggestion
| starter-url | Mention the gatsby starter pack url or local path here. See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more. Default starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default). |
```

default is here not a parameter or command, so i would leave it as normal text

### `new`

> ```bash

Author: > We don't specify these commands anywhere else on this document so I would skip it to keep it consistent.

Didnt get this one !

> ```bash

Reviewer 2: I'm not 100% sure we should show a shell snippet on how we use this as we don't do it anywhere. That's all. I'm fine with keeping this.

> ```bash

Author: > I'm not 100% sure we should show a shell snippet on how we use this as we don't do it anywhere

I guess we can use any of the two option below :
- can try changing most of these snippets to syntax highlighters
- I will remove the `bash` from this one and can keep others as it is.

Either options looks fine to me. Cause having these highlighters can be good and confusing at same time. Like they can be really good in understanding the code better , they can improve the code readability. But if some code not understand by the github markdown, they can be difficult to read. cause mostly in js codes, some new code style or new staged feature may not be handle by the markdown and may effect the highlighting.

> ```bash

Reviewer: We always use `shell` or `bash` on code blocks. I think emoving it is not the intention of the comment. 

I think the idea of the comment was: why the changes done only on the new command and not on all commands on this readme

> ```bash

Author: Ohh my bad then, yeah I will surely change it to other snippets as well

> ```bash

Author: @wardpeet Marking this as resolved . Please have it review again as I guess I changed all to same. Let me know if you want anything else to change.

gatsby new [<site-name> [<starter-url>]]
```

#### Arguments

> - **`site-name` :** Your Gatsby site name along with the path should be pass here`

Reviewer 2: We use tables in the rest of the document so I thought we might want it here too.

```
| Argument | Description |
|-------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| site-name | Your Gatsby site name along with the path should be pass here |
| starter-url | Mention the gatsby starter pack url or local path here. See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more. `Default` starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default). |
```

Reviewer: Good idea with the table

Author: I had this idea but then I thought that table is for the command options so I dropped it . Ok i will change it !

Author: done !

- **`starter-url` :** Mention the gatsby starter pack url or local path here.
  See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more.
  `Default` starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default).

> > **NOTE :** Dont pass `.` or `./` or `` _(space)_ in the `site-name`. It will throw error of specifying name and it will stop the operation

Reviewer 2: I feel like full-on sentences are bit easier to read.

```suggestion
> Note: The `site-name` should only consist of letters and numbers. If you specify a `.`, `./` or a `<space>` in the name, `gatsby new` will throw an error.
```

Author: Nice 👍
